File: cpp/387_First_Unique_Character_In_A_String/387_First_Unique_Character_In_A_String.cpp
#include<cstring>
#include<iostream>

using namespace std;

class Solution {
public:
	int firstUniqChar(string s) {
		int cnt[26] = { 0 };
		for (auto c : s) {
			cnt[c - 'a']++;
		}
		for (int i = 0; i < s.size(); i++) {
			if (cnt[s[i] - 'a'] == 1) return i;
		}
		return -1;
	}
};

File: python/387. First Unique Character in a String.py
class Solution:
    def firstUniqChar(self, s):
        s_dict = dict()
        for i,ch in enumerate(s):
            s_dict[ch] = i
        for i,ch in enumerate(s):
            if ch in s_dict and i==s_dict[ch]:
                return i
            elif ch in s_dict:
                s_dict.pop(ch)
        return -1



s = "lleetcode"
ExpectOutput = 4

solu = Solution()#先对类初始化，才能进行调用
temp = solu.firstUniqChar(s)
if (temp == ExpectOutput):
    print('right')
else:
    print('wrong')
    print(temp)